### PR TITLE
Update middlewares.md

### PR DESCRIPTION
**Pull request by mhkolk**
# Description

Adding a note informing developers to use overloads of Add method for middleware lifetime

### CODE REVIEW

**Review by brmagadutra — APPROVED** [review]

**Review by simaoribeiro — APPROVED** [review]

File: website/docs/guides/middlewares/middlewares.md [hunk 1]
```diff
@@ -17,6 +17,12 @@ Every product/consumer has its own Middlewares instances, so the instances are n
 
 Middlewares are instantiated by the configured dependency injection container, so every dependency configured by your container can be delivered in the middleware constructor.
 
+:::info
+
+Use overloads of the `Add<TMiddleware>(MiddlewareLifetime)` method when registering middlewares with your DI container to control middleware instance lifetime.
```
Comment (joelfoliveira) [inline website/docs/guides/middlewares/middlewares.md:22]: Hi @mhkolk,

Thank you for improving the documentation. 

Would you mind creating a section specific for this information in this page? Maybe naming it like "Configure Middleware Lifetime", similar to the one on [TypedHandlers](https://farfetch.github.io/kafkaflow/docs/guides/middlewares/typed-handler-middleware#configuring-handler-lifetime)

There is more useful information that we can share here:
- The available lifetimes ([here](https://github.com/Farfetch/kafkaflow/blob/master/src/KafkaFlow.Abstractions/MiddlewareLifetime.cs))
- The default lifetime (ConsumerOrProducer)
- Code sample on how to configure the middleware lifetime

Comment (mhkolk) [reply website/docs/guides/middlewares/middlewares.md:22]: This has been opened for a while and I have to say I don't have the time atm to expand what I already added.

Comment (joelfoliveira) [reply website/docs/guides/middlewares/middlewares.md:22]: No problem @mhkolk, we will merge as-is, and then we can open a new PR with the changes proposed.

Thank you for your contribution.

**Review by joelfoliveira — APPROVED** [review]

**Comment by joelfoliveira** [comment]
@mhkolk,

The pipeline is failing due to the commit message not following the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) format as per the [contribution guides](https://github.com/Farfetch/kafkaflow/blob/master/CONTRIBUTING.md).

Can you please amend the commit message and squash both commits (when trying to rebase the branch, GitHub created a merge commit).

**Review by JoaoRodriguesGithub — APPROVED** [review]

**Comment by JoaoRodriguesGithub** [comment]
Hi @mhkolk,

Since you did not address the requested changes so far, within the next 5 business days we will close this PR and create a new one in order to move forward. 

We appreciate your contribution to our project and look forward to hearing from you soon. If you have any questions or need further assistance, don't hesitate to reach out.

**Comment by JoaoRodriguesGithub** [comment]
Hi @mhkolk,

We will be closing this pull request.

Please feel free to respond to this message or reopen the issue if you'd like to continue the discussion or if you've encountered any new developments related to it.

Thank you for your contribution.

**Pull request closed without merging**